src/arch/x86/interrupts.cc

```diff
@@ -279,7 +279,7 @@ X86ISA::Interrupts::setThreadContext(ThreadContext *_tc)
 
     BaseInterrupts::setThreadContext(_tc);
 
-    initialApicId = tc->cpuId();
+    initialApicId = (tc->cpuId()  << 1) + _tc->threadId();
```

Comment:
We need to sync up with @takekoputa on this. I think that he and I had a citation for the "right" thing to do, but I don't remember.

Reply:
We still need to do this. What is the id supposed to be? Is it OK to have just 1 bit?

Reply:
This still needs to be addressed!

Reply:
@takekoputa: Do you remember where we found the "right" solution? I thought we had a reference for how to do this correctly, but I'm having trouble finding it again.